File: my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

class AddTwoIntsClientNode : public rclcpp::Node
{
public:
    AddTwoIntsClientNode() : Node("add_ints_two_client")
    {
        threads_.push_back(
            std::thread(std::bind(&AddTwoIntsClientNode::callbackAddTwoIntsService,this,1,4)));
        threads_.push_back(
            std::thread(std::bind(&AddTwoIntsClientNode::callbackAddTwoIntsService,this,3,6)));
    }

    void callbackAddTwoIntsService(int a,int b){
        auto client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
        while(!client -> wait_for_service(std::chrono::seconds(1))){
            RCLCPP_WARN(this->get_logger(),"Waiting for the server to be up...");
        }
        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a = a;
        request->b = b;
        auto future = client->async_send_request(request);
        try
        {
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(),"%d + %d = %ld",a,b,response->sum);
        }
        catch(const std::exception& e)
        {
            RCLCPP_ERROR(this->get_logger(),"Service call failed.");
        }
        
    }

private:
    std::vector<std::thread> threads_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<AddTwoIntsClientNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
from functools import partial

class AddTwoIntsClientNode(Node):

    def __init__(self):
        super().__init__("add_two_ints_client")
        self.callAddTwoIntsServer(6,7)

    def callAddTwoIntsServer(self,a,b):
        client = self.create_client(AddTwoInts,"add_two_ints")
        while not client.wait_for_service(1.0):
            self.get_logger().warn("Wait for server Add Two Ints...")

        request = AddTwoInts.Request()
        request.a = a
        request.b = b

        future = client.call_async(request)
        future.add_done_callback(partial(self.callbackAddTwoInts,a=a,b=b))

    def callbackAddTwoInts(self,future,a,b):
        try:
            response = future.result()
            self.get_logger().info(str(a) + " + " + str(b) + " = " + str(response.sum))
        except Exception as e:
            self.get_logger().error("Service call failed %r"%(e,))


def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClientNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()